cmd/platform/deploy.go

```diff
@@ -163,16 +163,24 @@ func hasValidDeploymentMethod(
 		return err
 	}
 	switch {
+	// When the manifest source is local, we can get the manifest from the local project.
 	case manifestSource.Equals(config.ManifestSourceLocal):
 		manifest, err = clients.AppClient().Manifest.GetManifestLocal(ctx, clients.SDKConfig, clients.HookExecutor)
 		if err != nil {
 			return err
 		}
-	case manifestSource.Equals(config.ManifestSourceRemote):
+	// When the manifest source is remote and the app exists, we can get the manifest from the the API.
+	case manifestSource.Equals(config.ManifestSourceRemote) && app.AppID != "":
```

Comment:
note: We need to differentiate between a remote manifest **with** an app (get manifest from app settings) and a remote manifest **without** an app (get manifest from local file).